testsuite/classlibrary/TestCoreUGens.sc

```diff
@@ -619,5 +619,43 @@ TestCoreUGens : UnitTest {
 
 	}
 
+	test_K2A_ramp_input {
+		var buf, node, flag = false;
+		var data;
+		var cond = CondVar.new;
 
+		server.bootSync;
+
+		buf = Buffer.alloc(server, server.options.blockSize * 16, 2);
+		server.sync;
+
+		node = {
+			var pulse = LFPulse.kr(ControlRate.ir * 0.25);
+			RecordBuf.ar(K2A.ar(pulse, [0, 1]), buf, loop: 0, doneAction: 2);
```

Comment:
you can also call `.asBuffer` directly on the function. That might be shorter.

Reply:
or even more directly, `.loadToFloatArray` can be called on a function.